addons/purchase_stock/models/purchase_order.py

```diff
@@ -78,6 +79,15 @@ def _compute_receipt_status(self):
     def _compute_dest_address_id(self):
         self.filtered(lambda po: po.picking_type_id.default_location_dest_id.usage != 'customer').dest_address_id = False
 
+    @api.depends('company_id')
+    def _compute_picking_type_visible(self):
+        if self.env.user.has_group('stock.group_stock_multi_locations'):
+            self.picking_type_visible = True
+            return
+        for company, orders in groupby(self, lambda o: o.company_id):
+            picking_type = self._get_picking_type(company.id)
+            self.env['purchase.order'].concat(*orders).picking_type_visible = not bool(picking_type)
```

Comment:
```suggestion
        for company, orders in self.grouped('company_id').items():
            picking_type = self._get_picking_type(company.id)
            orders.picking_type_visible = not bool(picking_type)
```
Maybe a bit easier to read this way?